dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -239,8 +236,13 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                 }
             );
 
-            if (WI_IsFlagSet(registrationOptions, PushNotificationRegistrationOptions::ComActivator))
+            if (WI_IsFlagSet(registrationActivators, PushNotificationRegistrationActivators::ComActivator))
             {
+                {
+                    auto lock = s_activatorInfoLock.lock_exclusive();
+                    THROW_HR_IF_MSG(E_INVALIDARG, s_cookie, "ComActivator already registered.");
+                }
+
                 GetWaitHandleForArgs().create();
 
                 THROW_IF_FAILED(::CoRegisterClassObject(
```

Comment:
You should just call `s_cookie.reset()`. Please look in ReSearch how this wil wrapper is used.